sound/soc/sof/xtensa/core.c

```diff
@@ -110,7 +110,7 @@ static void xtensa_stack(struct snd_sof_dev *sdev, void *oops, u32 *stack,
 			 u32 stack_words)
 {
 	struct sof_ipc_dsp_oops_xtensa *xoops = oops;
-	u32 stack_ptr = xoops->stack;
+	u32 stack_ptr = xoops->plat_hdr.stackptr;
```

Comment:
Does it work? It looks like `stackptr` is in dsp address space (stack top position in the moment of panic). Stack values copied to the memory window are located at `stackoffset`. Sorry for naming, idk why they are named so poorly (and why these names are alllowercaselikeinluaapi?).

Reply:
@kv2019i let me know if an update is needed to address @jajanusz ' question, will wait until tomorrow before merging

Reply:
@jajanusz @plbossart This is intentional. This value is only used to pretty print the results. Considering all the register contents is reflecting values from DSP address space, it would seem more useful to use DSP addresses in the stack print out. Address where the stack is copied in kernel space does not seem very useful information. 

So I think this is good. In any case for deeper debug, you will want to look at full "exception" file and parse it with core-dumper. Chime in @lgirdwood and @xiulipan but I think this is good to go. We can improve the stack dump later.

Reply:
@kv2019i Thanks. Then it's ok.